src/cockpit/peer.py

```diff
@@ -133,11 +133,13 @@ def eof_received(self) -> bool:
         # an EOF, then we consider it to be an error.  This allows us to
         # distinguish close caused by unexpected EOF (but no errno from a
         # syscall failure) vs. close caused by calling .close() on our side.
-        self.close(EOFError('The peer unexpectedly sent EOF'))
+        # The process is still running at this point, so keep it and handle
+        # the error in process_exited().
+        logger.debug('Peer %s received unexpected EOF', self.__class__.__name__)
```

Comment:
Why not just drop this vfunc override and set the flag on the underlying protocol object and let it get handled there?  This is precisely the case that this flag exists for.

Reply:
I'd love to, but I don't know how. Can you please expand on this? Which flag? If you mean `close_on_eof`, that's not a src/cockpit/protocol.py thing.

Reply:
Oh lol.  Oops.  I messed up reading this morning :rofl:.